var/spack/repos/builtin/packages/py-build/package.py

```diff
@@ -12,15 +12,21 @@ class PyBuild(PythonPackage):
     homepage = "https://github.com/pypa/build"
     pypi = "build/build-0.7.0.tar.gz"
 
+    version("0.10.0", sha256="d5b71264afdb5951d6704482aac78de887c80691c52b88a9ad195983ca2c9269")
+    version("0.9.0", sha256="1a07724e891cbd898923145eb7752ee7653674c511378eb9c7691aab1612bc3c")
+    version("0.8.0", sha256="887a6d471c901b1a6e6574ebaeeebb45e5269a79d095fe9a8f88d6614ed2e5f0")
     version("0.7.0", sha256="1aaadcd69338252ade4f7ec1265e1a19184bf916d84c9b7df095f423948cb89f")
 
     variant("virtualenv", default=False, description="Install optional virtualenv dependency")
 
     depends_on("python@3.6:", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:", when="@0.10:", type=("build", "run"))
+    depends_on("py-flit-core@3.4:", when="@0.10:", type="build")
+    depends_on("py-setuptools", when="@:0.9", type="build")
     depends_on("py-packaging@19:", type=("build", "run"))
-    depends_on("py-pep517@0.9.1:", type=("build", "run"))
+    depends_on("py-pep517@0.9.1:", when="@:0.9", type=("build", "run"))
     depends_on("py-tomli@1:", type=("build", "run"))
+    depends_on("py-tomli@1.1:", when="@0.10:", type=("build", "run"))
     depends_on("py-colorama", when="platform=windows", type=("build", "run"))
     depends_on("py-importlib-metadata@0.22:", when="^python@:3.7", type=("build", "run"))
     depends_on("py-virtualenv@20.0.35:", when="+virtualenv", type=("build", "run"))
```

Comment:
Can you check to see if this patch is still needed and update if so?

Reply:
Ah, that again. I recall the reason for that patch now, but not exactly how to trigger the issue.

In any case:
- the current patch, introduced in #28562 to address [pypa/build #266](https://github.com/pypa/build/issues/266), does not apply to any other versions,
- there were multiple PRs addressing [pypa/build #266](https://github.com/pypa/build/issues/266) that were opened/closed with lots of discussion, but the issue itself remains open.

Reply:
I can't find any reason to think this has been fixed. I'll extend the patch for all versions as a filter_file.